### PR TITLE
Added OpenShift documentation

### PR DESCRIPTION
**Pull request by tzumainn**

### CODE REVIEW

File: docs/source/usage/openshift.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
+OpenShift
+=========
+
+Owners and lessees can easily install OpenShift on hardware managed through ESI through the use of the `OpenShift assisted installer`_. This guide assumes that the relevant roles have been granted access to the required Ironic actions.
```
Comment (larsks) [inline docs/source/usage/openshift.rst:4]: I think this should use title caps: "OpenShift Assisted Installer" (here and throughout the document)

File: docs/source/usage/openshift.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+   * You may use https://console.redhat.com or a local installation of the OpenShift assisted installer
+* An HTTP server for the OpenShift ISO generated by the assisted installer
+* Usage of ESI-managed nodes in the 'available' state
+
+The Ironic provisioning network must also be able to reach the HTTP server and the assisted installer. If you are using https://console.redhat.com, the following commmands will provide the appropriate access:
+
+  .. prompt:: bash $
+
+    openstack subnet set --dns-nameserver 8.8.8.8 <provisioning subnet>
+    sudo iptables -t nat -A POSTROUTING -s <provisioning subnet cidr> -j MASQUERADE
```
Comment (larsks) [inline docs/source/usage/openshift.rst:19]: It's important to note that while the `openstack subnet ...` command can be run anywhere with access to the OpenStack API server, the `iptables` command must be run *on the ironic controller* (possibly controllers, plural, in an HA setup).

Comment (larsks) [reply docs/source/usage/openshift.rst:19]: Upon reflection, this is a bit of a hack: someone interacting with ESI as a service isn't necessarily going to have access to the controller and may not be able to make the necessary network changes.

File: docs/source/usage/openshift.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+* Wait for the nodes to appear on the assisted installer as Ready. Click Next.
+* In the 'Networking' step, select the appropriate subnet. Click Next.
+   * You can switch your node's networks if needed by running these commands and waiting for the subnet list to refresh:
+
+  .. prompt:: bash $
+
+    openstack esi node network list    # find names of ports attached to nodes
+    openstack esi node network detach <node> <port>
+    openstack esi node network attach --network <other network> <node>
+* Installation will continue and eventually complete. Once it does, the assisted installer will have credentials for logging into your OpenShift console.
```
Comment (larsks) [inline docs/source/usage/openshift.rst:68]: I think you're missing the steps necessary to actually start the install (in the assisted installer ui)? Your last instruction is "In the 'Networking' step, select the appropriate subnet. Click Next.", and I think there's something after that before the install actually starts.

File: docs/source/usage/openshift.rst (hunk 1 above)
Comment (tzumainn) [reply docs/source/usage/openshift.rst:4]: I actually thought so too, but the blog post doesn't so I was confused. I like caps better though, so I'll change it.

File: docs/source/usage/openshift.rst (hunk 2 above)
Comment (tzumainn) [reply docs/source/usage/openshift.rst:19]: Well, both are operator-level actions. I'll clarify in the doc that this is something like an installation step.

File: docs/source/usage/openshift.rst (hunk 3 above)
Comment (tzumainn) [reply docs/source/usage/openshift.rst:68]: Actually, there isn't! Clicking "Next" after selecting the subnet will start the installation.

**Comment by tzumainn** [comment]
@larsks Thanks for the review! I've addressed the comments, I think.

**Review by DanNiESh — APPROVED** [review]
Looks great! I don't see any problems.

**Review by larsks — APPROVED** [review]
Looks good to me.

**Pull request merged**